server/server.js

```diff
@@ -0,0 +1,115 @@
+const express = require('express');
+const users = require('../data/helpers/userDb');
+
+const server = express();
+server.use(express.json());
+
+const sendErrorMsg = (errCode, msg, res) => {
+    res.status(errCode);
+    res.json({ Error: msg });
+}
+
+// custom middleware
+
+const upperCase = (req, res, next) => {
+    console.log(req.params)
+    req.body.name = req.body.name.toUpperCase();
+
+    next();
+};
+
+//update user
+
+server
+    .put('/api/users/:id', upperCase, (req, res) => {
+        const { id } = req.params;
+        const changes = req.body;
+
+        users
+            .update(id, changes)
+            .then(count => {
+                if (count) {
+                    res.status(200).json({ message: `${count} name was updated` })
+                } else {
+                    return sendErrorMsg(400, 'id not found')
+                }
+            })
+            .catch(err => {
+                return sendErrorMsg(500, 'The name could not be modified');
+            });
+    });
+
+
+
+
+
+//get users
+
+server
+    .get('/', (req, res) => {
+        res.status(200).json({ api: 'running' });
+    });
+
+server.get('/api/users', (req, res) => {
+    users
+        .get()
+        .then(user => {
+            res.status(200).json(user);
+        })
+        .catch(err => {
+            return sendErrorMsg(500, 'Users information could not be retrieded', res)
+        });
+});
+
+server
+    .get('/api/users/:id', (req, res) => {
+        const { id } = req.params;
+
```

Comment:
Just a recommendation, the space in between post const and users makes the endpoint a little difficult to read and doesn't look clean.  It could be the github file that I'm looking at so if that is the case, disregard.